src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -390,7 +390,7 @@ Status GcsActorManager::RegisterActor(
   }
 
   auto actor = std::make_shared<GcsActor>(request);
-  if (actor->IsDetached()) {
+  if (actor->IsDetached() && !actor->GetName().empty()) {
```

Comment:
We need to handle named actors and detached actors separately. (This means that we also need to decouple these 2 concepts from the API.) I discussed this @edoakes and @ericl before.

Reply:
For now, named actor is detached actor. We also removed `detached=True` flag. https://docs.ray.io/en/latest/advanced.html#detached-actors

When did you discuss with them last time? Maybe there's some miscommunication?